prow/flagutil/github.go

```diff
@@ -84,9 +84,7 @@ func (o *GitHubOptions) Validate(bool) error {
 	}
 
 	if o.TokenPath == "" && o.AppID == "" && !o.AllowAnonymous {
-		// TODO(fejta): just return error after May 2020
-		logrus.Warnf("missing required flag: please set to --github-token-path=%s before June 2020", DefaultGitHubTokenPath)
-		o.TokenPath = DefaultGitHubTokenPath
+		return fmt.Errorf("missing required flag: please set to --github-token-path=%s", DefaultGitHubTokenPath)
```

Comment:
I'm not sure that this is appropriate any longer.

Components like crier, deck, mkpj are all used in githubless environments and should not require `--github-token-path != ""`

Reply:
alright, then should we close this and only remove the TODO?

Reply:
Well I do think we should remove `DefaultGitHubTokenPath` but not return an error if it is empty as some binaries (crier) sometimes but not always use github.

Reply:
Alright, but then when an app that needs it doesn't provide the option, we get such error message:
`error starting secret agent : error reading : open : no such file or directory`

Reply:
Maybe we should specify somehow that this config is optional for (crier) and others?

Reply:
Making the GitHubOptions struct more complicated so it can distinguish between token path being required or optional works for sure!

I'm okay them getting an `error stating secret agent` -- likely their path is not the default one so they set it anyway and even if they are using the default they've a super long time to start explicitly setting it.

Reply:
Alright, I chose the easiest of the 2 :)